internal/shared/controllers/pull_secret_controller.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// PullSecretReconciler reconciles a specific Secret object
+// that contains global pull secrets for pulling Catalog images
+type PullSecretReconciler struct {
+	client.Client
+	SecretKey                 *types.NamespacedName
+	ServiceAccountKey         types.NamespacedName
+	ServiceAccountPullSecrets []types.NamespacedName
+	AuthFilePath              string
+}
+
+func (r *PullSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithName("pull-secret-reconciler")
+
+	logger.Info("processing event", "name", req.NamespacedName)
+	defer logger.Info("processed event", "name", req.NamespacedName)
+
+	secrets := []*corev1.Secret{}
+	secret := &corev1.Secret{}
+
+	if r.SecretKey != nil { //nolint:nestif
```

Comment:
Might be nice to put this block in a separate function like `secret, err := r.getSecret(ctx, secretKey)`

Then we could reuse that function for both the global pull secret and the service account secrets, and it would probably no longer run afoul of the nestif linter.